ocaml/xapi/xapi_vm_lifecycle.ml

```diff
@@ -279,18 +279,28 @@ let check_protection_policy ~vmr ~op ~ref_str =
  * If the VM_metrics object does not exist, it implies the VM is not
  * running - in which case we use the current values from the database.
  **)
-let is_mobile ~__context vm strict =
+
+let nomigrate ~__context vm =
+  let metrics = Db.VM.get_metrics ~__context ~self:vm in
```

Comment:
We now get the metrics and platform (below) field twice for each VM. Can we get these in `is_mobile` and pass them to the two new functions (rather than the `vm`)?